go/arrow/extensions/bool8.go

```diff
@@ -0,0 +1,196 @@
+package extensions
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/apache/arrow/go/v17/arrow"
+	"github.com/apache/arrow/go/v17/arrow/array"
+	"github.com/apache/arrow/go/v17/arrow/memory"
+	"github.com/apache/arrow/go/v17/internal/json"
+)
+
+const ExtensionNameBool8 = "bool8"
```

Comment:
any particular reason for this to be exported?

Reply:
There is no reason currently, but I did have a thought about how it could be used.

Right now we have the functions `(Register|Unregister|Get)ExtensionType` for interacting with the global extension registry which developers can use to opt in to any extensions their application recognizes. As we make a clearer distinction between _regular_ extension types and canonical extension types, I was thinking that we could automatically pre-register extensions that are canonical so that applications automatically recognize them.

Whether we do this or not, the exported name key could make it easier to interact with the registry using a consistent identifier.

Thoughts?

Reply:
hmm, i think that's fine, not sure about automatically registering canonical types though. Does the C++ lib do that?

Reply:
It appears it does actually: https://github.com/apache/arrow/blob/main/cpp/src/arrow/extension_type.cc#L144-L154

Reply:
I removed this const entirely because it's only used in one place (in the `Bool8Type.ExtensionName` impl) and it's simple enough to just get the type itself, which is exported, and just call `ExtensionName()` if needed.

I'll also leave out auto-registering canonical extensions for now. Once Opaque merges and UUID gets moved out of `internal`, we can set that up for them together.

FYI all subsequent code changes are in #43323